### PR TITLE
Update wildcard enum match lint for non_exhaustive enums

### PR DESCRIPTION
**Pull request by illicitonion**
changelog: wildcard_enum_match_arm gives better suggestions for non_exhaustive enums

### CODE REVIEW

File: clippy_lints/src/matches.rs [hunk 1]
```diff
@@ -543,11 +543,20 @@ fn check_wild_enum_match(cx: &LateContext<'_, '_>, ex: &Expr, arms: &[Arm]) {
             return;
         }
 
+        let mut message = "wildcard match will miss any future added variants.";
```
Comment (lzutao) [inline clippy_lints/src/matches.rs]:
```suggestion
        let mut message = "wildcard match will miss any future added variants";
```

File: clippy_lints/src/matches.rs [hunk 2]
```diff
@@ -543,11 +543,20 @@ fn check_wild_enum_match(cx: &LateContext<'_, '_>, ex: &Expr, arms: &[Arm]) {
...
+
+        if let ty::Adt(def, _) = ty.kind {
+            if def.is_variant_list_non_exhaustive() {
+                message = "match on non-exhaustive enum doesn't explicitly match all known variants.";
```
Comment (lzutao) [inline clippy_lints/src/matches.rs]:
```suggestion
                message = "match on non-exhaustive enum doesn't explicitly match all known variants";
```

File: clippy_lints/src/matches.rs [hunk 3]
```diff
@@ -543,11 +543,20 @@ fn check_wild_enum_match(cx: &LateContext<'_, '_>, ex: &Expr, arms: &[Arm]) {
             return;
         }
 
+        let mut message = "wildcard match will miss any future added variants";
+
+        if let ty::Adt(def, _) = ty.kind {
+            if def.is_variant_list_non_exhaustive() {
+                message = "match on non-exhaustive enum doesn't explicitly match all known variants";
+                suggestion.push(String::from("_"));
```
Comment (flip1995) [inline clippy_lints/src/matches.rs]: Why also push `_`, when the match is exhaustive with the suggested changes?

Comment (flip1995) [reply clippy_lints/src/matches.rs]: Oh `non_exhaustive` is a thing now, didn't realize that.

**Comment by flip1995** [comment]
Thanks!

@bors r+

**Comment by flip1995** [comment]
It seems that another rustup is necessary...

**Pull request merged**